oap-ape/ape-java/ape-flink/src/main/java/org/apache/flink/yarn/ApeYarnResourceManagerDriver.java

```diff
@@ -376,10 +383,25 @@ private ContainerLaunchContext createTaskExecutorLaunchContext(
 		final ContaineredTaskManagerParameters taskManagerParameters =
 			ContaineredTaskManagerParameters.create(flinkConfig, taskExecutorProcessSpec);
 
-		log.info("TaskExecutor {} will be started on {} with {}.",
+		// select numa node if configured
+		String logInfo = "";
+		List<String> numaNodeList = flinkConfig.get(ApeYarnOptions.NUMA_BINDING_NODE_LIST);
+		if (numaNodeList.size() > 0) {
+			int numaCount = numaNodeList.size();
+			numaBindingNodeCounter.putIfAbsent(host, new AtomicInteger(new Random().nextInt(numaCount)));
```

Comment:
hmm, whether this will cause some unbalance issue? Some worst cases, all random to the same NUMA node.

Reply:
This random is just to generate a start index.
Then the following task managers will be distributed in a balanced logic.
This random is designed to get rid of the bad effects when many Flink application start their 1st task manager on the first numa node.